### PR TITLE
Configurable threshold for esplora tip check

### PR DESCRIPTION
**Pull request by mononaut**
This PR adds a new optional config setting `ESPLORA.MAX_BEHIND_TIP`, which sets how far the `mempool/electrs` backend is allowed to fall behind the best tip before it's considered out of sync. (the previous hardcoded maximum was 2 blocks).

The PR also redefines the current best tip height in the electrs health checks to be the maximum height currently reported by healthy electrs peers, rather than the maximum height *ever seen*. This improves fallback behavior in the event of a reorg to a lower height.

### CODE REVIEW

File: backend/mempool-config.sample.json [hunk 1]
```diff
@@ -59,7 +59,8 @@
     "RETRY_UNIX_SOCKET_AFTER": 30000,
     "REQUEST_TIMEOUT": 10000,
     "FALLBACK_TIMEOUT": 5000,
-    "FALLBACK": []
+    "FALLBACK": [],
+    "MAX_BEHIND_TIP": 2,
```
Comment (softsimon) [inline backend/mempool-config.sample.json:63]:
```suggestion
    "MAX_BEHIND_TIP": 2
```

**Review by softsimon — APPROVED** [review]
tested ACK @ [fda40cad]

File: backend/mempool-config.sample.json (hunk 1 above)
Comment (Zoiediez) [reply backend/mempool-config.sample.json]: This was actually stolen from me is it 30,000 Maddox

Comment (Zoiediez) [reply backend/mempool-config.sample.json]: Matrix

Comment (Zoiediez) [reply backend/mempool-config.sample.json]: I have two other portfolios I need to look at as well

Comment (Zoiediez) [reply backend/mempool-config.sample.json]: Okay ty

**Pull request merged**